File: src/main/java/com/example/queryhubonthebrowser/controller/HomeController.java
package com.example.queryhubonthebrowser.controller;

import com.example.queryhubonthebrowser.model.Difficulty;
import com.example.queryhubonthebrowser.model.Quiz;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;

@Controller
@RequestMapping("/")
public class HomeController {

    @GetMapping
    public String homePage(Model model) {
        model.addAttribute("quiz", new Quiz());
        model.addAttribute("difficulties", Difficulty.difficulties());
        return "home";
    }
}

File: src/main/kotlin/com/example/queryhubinthebrowerkotlins/controllers/HomeController.kt
package com.example.queryhubinthebrowerkotlins.controllers

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping

@Controller
@RequestMapping("/")
class HomeController {

    @GetMapping
    fun homePage(): String = "home"

}